salt/modules/win_file.py

```diff
@@ -1035,6 +1035,68 @@ def set_mode(path, mode):
     return get_mode(path)
 
 
+def remove(path, force=False):
+    '''
+    Remove the named file or directory
+
+    :param str path: The path to the file or directory to remove.
+
+    :param bool force: Remove even if marked Read-Only
+
+    :return: True if successful, False if unsuccessful
+    :rtype: bool
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' file.remove C:\\Temp
+    '''
+    # This must be a recursive function in windows to properly deal with
+    # Symlinks. The shutil.rmtree function will remove the contents of
+    # the Symlink source in windows.
+
+    path = os.path.expanduser(path)
+
+    # Does the file/folder exists
+    if not os.path.exists(path):
+        return 'File/Folder not found: {0}'.format(path)
+
+    if not os.path.isabs(path):
+        raise SaltInvocationError('File path must be absolute.')
+
+    # Remove ReadOnly Attribute
+    if force:
+        # Get current file attributes
+        file_attributes = win32api.GetFileAttributes(path)
+        win32api.SetFileAttributes(path, win32con.FILE_ATTRIBUTE_NORMAL)
+
+    try:
+        if os.path.isfile(path):
+            # A file and a symlinked file are removed the same way
+            os.remove(path)
+        elif is_link(path):
+            # If it's a symlink directory, use the rmdir command
```

Comment:
Since `os.path.isfile()` will follow the link to its destination, this logic doesn't make sense. Either `is_link` should be first, or this whole logic should be reconsidered.